### PR TITLE
Allow ember-cli-simple-store to conditionally live along side ember-data

### PR DESCRIPTION
**Pull request by jasonmit**
I'm proposing the ability to conditionally disable the auto injection of store, as well as renaming the initializer.  This will allow ember-cli-simple-store to live along side ember-data, which is a circumstance we're dealing with while we need the two to live in harmony as we port the application over to ember-cli-simple-store.

Hopefully this will help others.  Thanks again for your hard work :+1:

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -8,7 +8,7 @@
     "ember-qunit": "0.4.9",
     "ember-qunit-notifications": "0.0.7",
     "ember-resolver": "~0.1.18",
-    "jquery": "^1.11.3",
+    "jquery": "1.11.3",
```
Comment (jasonmit) [inline bower.json:11]: Intentional https://github.com/ember-cli/ember-cli/commit/4cb6205015b443da0aabf36072a29f74723e09df

Comment (toranb) [inline bower.json:11]: the jQuery issue that plagues us all right now :)

**Comment by toranb** [comment]
I was actually thinking about this recently :) for the "incremental upgrade" scenario I'm sure you are facing today

**Comment by toranb** [comment]
@jasonmit Any chance you could add something about this config option in the readme for others who stop by?

**Comment by jasonmit** [comment]
Will do, and I was planning to do one of two things on how to register/inject the simple-store.  Either by making simple store an Ember.Service and document Ember.injectservice or documenting setting up an initializer.  Which do you prefer?

**Comment by jasonmit** [comment]
Or I can do both if you want to maintain support pre Ember.Service

File: README.md [hunk 2]
```diff
@@ -219,6 +219,23 @@ animal.set("omnivore", false);
 animal.get("isDirty"); //false
 ```
 
+## Disabling injection
```
Comment (jasonmit) [inline README.md]: I still think we should expose ember-cli-simple-store as a service so consumers can `Ember.inject.service`.  The alternative is documenting setting up a new initializer, which not many people are familiar with.

Thoughts?  I can include that in this PR as well and update the README.

**Comment by jasonmit** [comment]
Updated the documentation with instructions on `disableAutoInject` and re-registering `ember-cli-simple-store/store`.

**Comment by toranb** [comment]
@jasonmit sorry for the lack of response the last 24 hours but I've got a conference talk today I was preparing for :)

The idea of a service has been thrown around and it matches the direction ember-data has taken. But because I haven't used ember-data in some time ... it might be worth asking you what pros/cons you see of "yes, lets add a service and that's how you get at store moving forward because adding another injection addon just to do what ember has out of the box isn't needed any longer" or "no, having it injected with something else has been great"

**Comment by toranb** [comment]
@jasonmit also -the docs look great! thank you!

One thing to consider is that I've been preparing a v4.0 release the last few weeks and this change should be in the 3.x series (along w/ the other PR that @thoov has up current for a documentation fix).

I'm planning to pull the v4.0 (beta) commits off master tomorrow and throw them into a branch so we can do the right thing and publish a v3.6.0 tomorrow w/ these configuration changes you need.

Is it okay if I pull those off and have you resubmit this (after I rewrite history by removing the v4.0 code on master)? Unless you are cool with waiting a bit for something like the v4.0 release ...

**Comment by jasonmit** [comment]
@toranb feel free to do what you need on master, I can adjust everything as this is likely better suited for 4.0.0 anyway if we go the Ember.Service approach.

As for using an Ember.Service, I'm all for it for a few reasons.  Clean API, future proof, and gives consumers more control over the property name to avoid collisions we see between store and simpleStore.

``` js
export default Ember.Route.extend({
  // I have to verify this, but if you intent to stick with `store` as the service name
  // then the work around we would document is something like:
  simpleStore: Ember.inject.service('ember-cli-simple-store/store')
});
```

It's not very different than what you have now, a singleton Ember.Object.  The only difference between an Ember.Object and an Ember.Service is the location of the service so that the resolver can locate the service and a special property on it https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/system/service.js#L45

You can still inject the service into common factory types, via an initializer like what you're doing now, and I can convert the logic I have to conditionally disable.  The one thing that will no longer be needed is the re-implementation of the initializer as we can just point users to the `Ember.inject.service` API.  It's a nice clean and friendly API that should remain future proof.

**Comment by toranb** [comment]
@jasonmit any chance you want to join up as a core contributor ? As the adoption grows it might be nice to have someone else to bounce ideas off /etc

**Comment by jasonmit** [comment]
@toranb I would love to help out.  We'll be using it heavily, so it would be great to be more closely involved

**Comment by toranb** [comment]
@jasonmit just added you to the github repo and npm :)

```
npm owner add jasonmit ember-cli-simple-store
```

**Comment by jasonmit** [comment]
:+1: thanks :)

**Comment by toranb** [comment]
@jasonmit I like the service idea- can we track that in another issue? I'd also like to have that be part of the v4.0.0 release planned later this month/early next. That way any "breaking" changes (hopefully very few) are easier to swallow

On this issue -the only real discussion/challenge is that I have v4.0 code on master (but you mention you might just wait and use it as-is or did you want this back in the 3.x series) ?

**Comment by toranb** [comment]
@jasonmit to be clear -the 3.x is super solid and the 4.x has a memory leak (currently why it's beta.5 and not a true stable v4.0 release at the moment)

**Comment by jasonmit** [comment]
If you're fine with this, then this can be part of the 3.x series since it's not a breaking change and will unblock some people porting there app over incrementally.  We can discuss the Service and potential breaking changes in an issue that I'll create and plan for it in 4.x

Ping me when you rewrite history/revert the commits you're looking to remove on master and I'll rebase this.

**Comment by toranb** [comment]
@jasonmit history has been dumped to a new branch called "branches/v4.0" and master is back to 3.x friendly so you can either rebase (if that will work) or submit a new PR :)

https://github.com/toranb/ember-cli-simple-store/commits/master

**Comment by toranb** [comment]
@jasonmit if rebase still shows > 2 commits can we do another PR (github is really confused right now showing a TON of commits)

**Comment by jasonmit** [comment]
Closing in favor of #35

**Pull request closed without merging**